lib.php

```diff
@@ -148,3 +152,91 @@ function poster_page_type_list($pagetype, $parentcontext, $currentcontext) {
         'mod-poster-view' => get_string('page-mod-poster-view', 'mod_poster'),
     );
 }
+
+/**
+ * Given a coursemodule object, this function returns the extra
+ * information needed to print this activity in various places.
+ *
+ * If poster needs to be displayed inline we store additional information
+ * in customdata, so functions {@link poster_cm_info_dynamic()} and
+ * {@link poster_cm_info_view()} do not need to do DB queries
+ *
+ * @param cm_info $cm
+ * @return cached_cm_info info
+ */
+function poster_get_coursemodule_info($cm) {
+    global $DB;
+    if (!($poster = $DB->get_record('poster', array('id' => $cm->instance),
+        'id, name, display, showdescriptionview, intro, introformat'))) {
+        return null;
+    }
+    $cminfo = new cached_cm_info();
+    $cminfo->name = $poster->name;
+    if ($poster->display == POSTER_DISPLAY_INLINE) {
+        // prepare poster object to store in customdata
+        $fdata = new stdClass();
+        $fdata->showdescriptionview = $poster->showdescriptionview;
+        if ($cm->showdescription && strlen(trim($poster->intro))) {
```

Comment:
Better to use `html_is_blank()` here, consistently with how the `view_page_description()` method does it